bpf/lib/conntrack.h

```diff
@@ -246,7 +246,8 @@ __ct_lookup(const void *map, struct __ctx_buff *ctx, const void *tuple,
 			ct_state->proxy_redirect = entry->proxy_redirect;
 			ct_state->from_l7lb = entry->from_l7lb;
 			ct_state->from_tunnel = entry->from_tunnel;
-			ct_state->ifindex = entry->ifindex;
+			if (!is_defined(HAVE_FIB_IFINDEX))
+				ct_state->ifindex = entry->ifindex;
```

Comment:
Let's also wrap the  `ct_state->ifindex` users in `nodeport_rev_dnat_ingress_ipv*()` ? Otherwise this looks like the `ifindex` wasn't even used anymore :).

And maybe also the `ifindex` field in the `ct_state` itself, similar as for `DISABLE_LOOPBACK_LB`.

Reply:
Agree, good point. ct_state is not part of map, so this is doable.